File: src/main/java/com/kkllffaa/meteor_litematica_printer/Functions/MathUtils.java
package com.kkllffaa.meteor_litematica_printer.Functions;

public class MathUtils {

    public static boolean isNearValue(float subject, float value, float halfRange) {
    	return MathUtils.isInOpenRangeValues(subject, value - halfRange, value + halfRange);
    }

    public static boolean isInOpenRangeValues(float subject, float min, float max) {
    	return subject > min && subject < max;
    }

    public static boolean acuteAngleSpansZero(float yaw1, float yaw2) {
    	return (Float.floatToIntBits(yaw1) ^ Float.floatToIntBits(yaw2)) < 0 && Math.abs(yaw2 - yaw1) < 180;
    }
    
}

File: src/main/kotlin/com/kkllffaa/meteor_litematica_printer/Functions/MathUtils.kt
package com.kkllffaa.meteor_litematica_printer.Functions

import java.lang.Float
import kotlin.Boolean
import kotlin.compareTo
import kotlin.math.abs

object MathUtils {
    fun isNearValue(subject: Float, value: Float, halfRange: Float): Boolean {
        return isInOpenRangeValues(subject, value - halfRange, value + halfRange)
    }

    fun isInOpenRangeValues(subject: Float, min: Float, max: Float): Boolean {
        return subject > min && subject < max
    }

    fun acuteAngleSpansZero(yaw1: Float, yaw2: Float): Boolean {
        return (Float.floatToIntBits(yaw1) xor Float.floatToIntBits(yaw2)) < 0 && abs(yaw2 - yaw1) < 180
    }
}
